cpp/include/cudf/join.hpp

```diff
@@ -212,7 +212,105 @@ std::unique_ptr<cudf::experimental::table> full_join(
                          std::vector<cudf::size_type> const& right_on,
                          std::vector<std::pair<cudf::size_type, cudf::size_type>> const& columns_in_common,
                          rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+/** 
+ * @brief  Performs a left semi join on the specified columns of two 
+ * tables (left, right)
+ *
+ * A left semi join only returns data from the left table, and only 
+ * returns rows that exist in the right table.
+ *
+ * @example TableA a: {0, 1, 2}
+ *          TableB b: {1, 2, 3}, a: {1, 2, 5}
+ *          left_on: {0}
+ *          right_on: {1}
+ *          return_columns: { 0 }
+ * Result: { a: {1, 2} }
+ *
+ * @example TableA a: {0, 1, 2}, c: {1, 2, 5}
+ *          TableB b: {1, 2, 3}
+ *          left_on: {0}
+ *          right_on: {0}
+ *          return_columns: { 1 }
+ * Result: { c: {1, 2} }
+ *
+ * @throws cudf::logic_error if either table is empty
```

Comment:
Shouldn't we be able to handle empty inputs w/o error?

Reply:
Clarified the comment.  The issue is if either table has no columns.  If the table has no rows the behavior is as expected.